README.md

```diff
@@ -67,6 +67,30 @@ your target contains `ibazel_notify_changes` in its `tags` attribute, then the
 command will stay alive and will receive a notification of the source changes on
 stdin.
 
+## Output Runner
+
+iBazel is capable of producing and running commands from the output of Bazel commands. If iBazel is run with the flag `--run_output` then it will check for a `%WORKSPACE%/.bazel_fix_commands.json` and if present run any commands that match the provided regular expressions.
+For example the commands defined by the following file will match `buildozer` commands found in the output and provide a prompt to run the command.
+```
+[{
+    "regex": "^buildozer '(.*)'\\s+(.*)$",
+    "command": "buildozer",
+    "args":    ["$1", "$2"],
+    "stopAfter": 127,
+}]
+```
+Adding the flag `--run_output_interactive=false` will automatically run the command without prompting for confirmation.
+The fields in `.bazel_fix_commands.json` are:
+
+* regex: a regular expression that will be matched against every line of output.
+    * backslash `\` characters will need to be escaped once for the regex to be parsed properly.
+* command: a command that will be run from the workspace root.
+* args: a list of arguments to provide to the command, with `$1` being the first match group of `regex`, `$2` being the second and so on.
+* stopAfter (optional): a number (0-127) of failures (non-0 exit codes from the `command`) after which to discard any remaining output.
```

Comment:
WRT `stopAfter` I'm trying to think of the way people are going to experience this. To my naive eye there are 2 modes that these generator tools operate. First you have the tools that emit a single build instruction like `gazelle` who rewrite all the `BUILD` files in the repo (that it knows how to write). You would configure something that invokes Gazelle.

```
{
    "regex": "^Please run gazelle$",
    "command": "bazel",
    "args":    ["run", "--", "-mode=fix"],
    "stopAfter": 1, // <-- This is 1 because running Gazelle multiple times will never help, so coalesce all the calls into a single invocation.
}
```

Where something like `buildozer` might have bunch of events that were emitted and you would configure it this way:

```
{
    "regex": "^buildozer '(.*)'\\s+(.*)$",
    "command": "buildozer",
    "args":    ["$1", "$2"],
    "stopAfter": 127, // <-- It will invoke this up to 127 times.
}
```

What isn't clear to me is how it is determining what to run up to 127 times. I could interpret this to mean that it will invoke `buildozer add dep //blah //foo` 127 times until it gets an exit status of `0`, or that it is going to put all the `buildozer` commands into a queue and run `127` of them in a row before giving up?

Reply:
I think there are 3 types of scenarios.
1 is the `gazelle` type fix. `stopAfter` is not helpful here.
2 is the `rules_scala_annex` type fix where it detects problems directly with the build file and outputs one error.  This leads to 1 error to 1 action mapping.  `stopAfter` is not helpful here.
3 is the `rules_typescript` type fix where the compiler identifies a missing a dependency, but then outputs an error for for each import of each file that is missing a dependency.  For the code base I am working on that involved about 3000 errors that mapped to 27 actions.  `stopAfter` is designed for this case.  The idea is that it will fix the first 3-4 errors, then try the compile step again, and hopefully only get 1000 errors, then do a few more actions, and so on until the compiler does not output any more things for the output_runner to handle.

In my case I use `"stopAfter": 3` to allow for some malformed commands, but to generally stop and restart as soon as possible.

So your second clause is the correct one, it will not try to parse the same output more than once, but if different output is matched by the same command regex, then it will only run it until it has failed `stopAfter` number of times. (It will in fact stop running all commands after 1 exceeds the `stopAfter` limit.)
If you have a way to phrase this in the documentation to make it more clear, let me know.